larray_editor/comparator.py

```diff
@@ -131,92 +132,52 @@ def display(self, diff_only):
         self.arraywidget.set_data(array, bg_value=bg_value)
 
 
-class ArrayComparator(QDialog):
+class ArrayComparator(AbstractEditor):
     """Array Comparator Dialog"""
     def __init__(self, parent=None):
-        QDialog.__init__(self, parent)
-
-        # Destroying the C++ object right after closing the dialog box,
-        # otherwise it may be garbage-collected in another QThread
-        # (e.g. the editor's analysis thread in Spyder), thus leading to
-        # a segmentation fault on UNIX or an application crash on Windows
-        self.setAttribute(Qt.WA_DeleteOnClose)
-
-    def setup_and_check(self, arrays, names, title=''):
-        """
-        Setup ArrayComparator:
-        return False if data is not supported, True otherwise
-        """
-        icon = ima.icon('larray')
-        if icon is not None:
-            self.setWindowIcon(icon)
-
-        if not title:
-            title = _("Array comparator")
-        title += ' (' + _('read only') + ')'
-        self.setWindowTitle(title)
+        AbstractEditor.__init__(self, parent)
+        self.setup_menu_bar()
+
+    def _setup_and_check(self, widget, data, title, readonly, **kwargs):
+        """Setup ArrayComparator"""
+        arrays = data
+        names = kwargs.get('names', ["Array{}".format(i) for i in range(len(arrays))])
 
         layout = QVBoxLayout()
+        widget.setLayout(layout)
+
         comparator_widget = ComparatorWidget(self)
         comparator_widget.set_data(arrays, Axis(names, 'array'))
         layout.addWidget(comparator_widget)
 
-        # Buttons configuration
-        btn_layout = QHBoxLayout()
-        btn_layout.addStretch()
+    def _setup_edit_menu(self, menu_bar):
+        pass
 
-        buttons = QDialogButtonBox.Ok
-        bbox = QDialogButtonBox(buttons)
-        bbox.accepted.connect(self.accept)
-        btn_layout.addWidget(bbox)
-        layout.addLayout(btn_layout)
-
-        self.setLayout(layout)
-        self.resize(800, 600)
-        self.setMinimumSize(400, 300)
+    def type_editor(self, readonly):
```

Comment:
I don't find the name of this method clear. What about editor_name? or editor_type?